### DISCUSSION TITLE
寻求指纹开发支持

### DISCUSSION THREAD

**Discussion by Garryz** [linuxdeepin/developer-center#3995] [Development Questions | 开发答疑]
我有一款指纹要接入系统，但只有so文件和头文件，有没有法子实现接入？

**Comment by BLumia** [comment]
你是开发者？（开发者也不至于只有 so 文件和文件头吧，是厂商只提供了二进制文件的意思吗？）

如果不是开发者的话，参见： https://github.com/orgs/linuxdeepin/discussions/3409

如果是开发者适配的话，实际就是 fprintd 的适配支持，和发行版是哪个的关系其实不大。

**Reply by Garryz** [reply]
fprint没有设备的id，厂商只提供了二进制文件

**Reply by CoelacanthusHex** [reply]
那你應該是需要用廠商提供的庫在 https://gitlab.freedesktop.org/libfprint/libfprint/-/tree/master/libfprint 下面編寫這個指紋掃描器的 fprint 驅動

**Reply by Garryz** [reply]
把so和进这个库，这样可行吗？

**Reply by CoelacanthusHex** [reply]
你還得添加交互的代碼，只是把這個庫放進去和啥也沒做沒區別

**Reply by Garryz** [reply]
我知道啊，但是把so放进开源库里，这样合适吗？

**Comment by Garryz** [comment]
https://blog.csdn.net/mr_zhaojy/article/details/122297445 照这个做了，还是不行

**Reply by LTreeshu** [reply]
你可以说下哪步出错了，有哪些报错提示，这样大家才能更好的知道问题在哪里，一起找解决办法

**Reply by Garryz** [reply]
我用go写了个空壳的程序，还没有接入驱动，怎么看报错，程序一直在跑，但是账户界面那里没有指纹的选项

**Reply by LTreeshu** [reply]
也就是说你写的配置文件没有被识别？控制中心里找不到可添加的指纹配置是吗？（教程中的步骤，配置文件和目录都有可能导致扫描不到）

**Reply by LTreeshu** [reply]
可以检查下目录和配置文件的书写

**Reply by LTreeshu** [reply]
还有一种情况是硬件型号并不支持指纹添加，例如华为笔记本

**Reply by Garryz** [reply]
控制中心的是哪个配置，为了确保有效，我已经把服务名和接口名统一成一个了，也检查过配置了，而且程序确实在启动的时候以root账号启动了，有没有什么地方可以看到日志，或者我程序去写日志

**Reply by Garryz** [reply]
硬件支持，厂家有个demo，可以识别到，但是只能用他的demo才能识别到，windows 的硬件管理识别不到，我是在虚拟机里装的deepin，里面也只有厂家的demo才识别的到

**Reply by LTreeshu** [reply]
还有一种就是系统还没安装fprind相关的软件包，可以搜索安装一下，我也并不是很了解这方面，提供一个大概的思路，希望对你有帮助，这里是我找到一个关于deepin的自定义规范https://blog.csdn.net/mr_zhaojy/article/details/122297043

**Reply by LTreeshu** [reply]
> 控制中心的是哪个配置，为了确保有效，我已经把服务名和接口名统一成一个了，也检查过配置了，而且程序确实在启动的时候以root账号启动了，有没有什么地方可以看到日志，或者我程序去写日志

deepin自带一个日志收集工具，你可以试试

**Reply by Garryz** [reply]
好的，谢谢

**Comment by zccrs** [comment]
fprintd没仔细看过，不过在它的代码仓库里放一个so文件大概率是不被允许的。如果你是想只适配deepin，我们有另外一套接口可以对接，可以不通过 fprintd 实现， 不过这个文档还没有公开，有需要可以询问 @black-desk

**Comment by Garryz** [comment]
@black-desk , 求指纹开发的文档

**Reply by black-desk** [reply]
文档整理后会发布在 https://docs.deepin.org/info/%E5%BC%80%E5%8F%91%E5%85%A5%E9%97%A8/%E5%9F%BA%E7%A1%80%E7%8E%AF%E5%A2%83/DTK/%E6%A6%82%E8%BF%B0/%E6%A6%82%E8%BF%B0/%E7%BB%9F%E4%BF%A1DTK 发布后会在这里再评论一下通知您

**Reply by zccrs** [reply]
> 文档整理后会发布在 https://docs.deepin.org/info/%E5%BC%80%E5%8F%91%E5%85%A5%E9%97%A8/%E5%9F%BA%E7%A1%80%E7%8E%AF%E5%A2%83/DTK/%E6%A6%82%E8%BF%B0/%E6%A6%82%E8%BF%B0/%E7%BB%9F%E4%BF%A1DTK 发布后会在这里再评论一下通知您

可以先在这里上传一份

**Reply by black-desk** [reply]
> > 文档整理后会发布在 https://docs.deepin.org/info/%E5%BC%80%E5%8F%91%E5%85%A5%E9%97%A8/%E5%9F%BA%E7%A1%80%E7%8E%AF%E5%A2%83/DTK/%E6%A6%82%E8%BF%B0/%E6%A6%82%E8%BF%B0/%E7%BB%9F%E4%BF%A1DTK 发布后会在这里再评论一下通知您
> 
> 可以先在这里上传一份

这份pdf里面带了很多统信的字眼和logo, 以社区的名义发布在这里并不合适.

**Reply by Garryz** [reply]
私聊？

**Reply by Garryz** [reply]
> > > 文档整理后会发布在 https://docs.deepin.org/info/%E5%BC%80%E5%8F%91%E5%85%A5%E9%97%A8/%E5%9F%BA%E7%A1%80%E7%8E%AF%E5%A2%83/DTK/%E6%A6%82%E8%BF%B0/%E6%A6%82%E8%BF%B0/%E7%BB%9F%E4%BF%A1DTK 发布后会在这里再评论一下通知您
> > 
> > 
> > 可以先在这里上传一份
> 
> 这份pdf里面带了很多统信的字眼和logo, 以社区的名义发布在这里并不合适.

大佬，要不你发到uos那边，我去那边找

**Reply by Garryz** [reply]
> > > 文档整理后会发布在 https://docs.deepin.org/info/%E5%BC%80%E5%8F%91%E5%85%A5%E9%97%A8/%E5%9F%BA%E7%A1%80%E7%8E%AF%E5%A2%83/DTK/%E6%A6%82%E8%BF%B0/%E6%A6%82%E8%BF%B0/%E7%BB%9F%E4%BF%A1DTK 发布后会在这里再评论一下通知您
> > 
> > 
> > 可以先在这里上传一份
> 
> 这份pdf里面带了很多统信的字眼和logo, 以社区的名义发布在这里并不合适.

大佬，我的空壳程序联通了，busctl接口都正常，但系统没有反应
